R/Connection.R

```diff
@@ -712,6 +712,53 @@ setMethod(
     invisible(TRUE)
   })
 
+#' List locations of ODBC configuration files
+#'
+#' @description
+#' On MacOS and Linux, odbc uses the unixODBC driver manager to manage
+#' information about driver and data sources. This helper returns the filepaths
+#' where the driver manager will look for that information.
+#'
+#' This function is a wrapper around the command line call `odbcinst -j`.
+#'
+#' Windows does not use `.ini` configuration files; this function will return a
+#' 0-length vector on Windows.
+#'
+#' @seealso
+#' The [odbcListDrivers()] and [odbcListDataSources()] helpers return
+#' information on the contents of `odbcinst.ini` and `odbc.ini` files,
+#' respectively.
+#'
+#' Learn more about unixODBC and the `odbcinst` utility
+#' [here](https://www.unixodbc.org/odbcinst.html).
+#'
+#' @examplesIf FALSE
+#' configs <- odbcListConfig()
+#'
+#' file.edit(configs[1])
+#' @export
+odbcListConfig <- function() {
+  if (identical(.Platform$OS.type, "windows")) {
+    return(character(0))
+  }
+
+  if (identical(unname(Sys.which("odbcinst")), "")) {
+    stop(
+      c("The unixODBC driver manager is not available. ",
+        "Please install and try again.")
+    )
+  }
+
+  res <- system("odbcinst -j", intern = TRUE)
+  res <- res[grepl("\\.ini", res)]
+  res <- strsplit(res, "\\:")
+  res <- vapply(res, `[[`, character(1), 2)
```

Comment:
I think it's worth considering a quick check of the output here, maybe something like this:

```R
if (!identical(length(res), c(2, 2, 2)) {
  abort("Failed to parse output from odbcinst", internal = TRUE)
}
```

That'll ensure the user gets a reasonable error if `odbcinst` gives unexpected results. (And the `internal = TRUE` ensures they get a prompt to file an issue).

Then you can add a snapshot test for that error by mocking `system()`.

Reply:
[275b69c](https://github.com/r-dbi/odbc/pull/680/commits/275b69c3740b695d8b96fb6e2d265d81553c1fa6).

Reply:
My only question here is whether we should run a similar check instead on the output of `system("odbcinst -j", intern = TRUE)`? Based on it's source code I'm skeptical that odbcinst will ever return something that's _almost_ parsable for us, but some issue in the communication with `system()` seems plausible.

Reply:
I don't think it's worth spending too much more time contemplating this as it's an unlikely failure mode; we just want to make sure we get something diagnostic in the unlikely event that the output does change.